browser/permissions/permission_manager_browsertest.cc

```diff
@@ -341,6 +341,42 @@ IN_PROC_BROWSER_TEST_F(PermissionManagerBrowserTest,
   }
 }
 
+IN_PROC_BROWSER_TEST_F(PermissionManagerBrowserTest,
+                       RequestPermissionsAllowETLDP1) {
+  struct {
+    std::vector<std::string> addresses;
+    ContentSettingsType type;
+  } cases[] = {{{"0xaf5Ad1E10926C0Ee4af4eDAC61DD60E853753f8A",
+                 "0xaf5Ad1E10926C0Ee4af4eDAC61DD60E853753f8B"},
+                ContentSettingsType::BRAVE_ETHEREUM},
+               {{"BrG44HdsEhzapvs8bEqzvkq4egwevS3fRE6ze2ENo6S8",
+                 "JDqrvDz8d8tFCADashbUKQDKfJZFobNy13ugN65t1wvV"},
+                ContentSettingsType::BRAVE_SOLANA}};
+
+  GURL top_url(https_server()->GetURL("a.test", "/iframe.html"));
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), top_url));
+  GURL iframe_url(https_server()->GetURL("c.b.a.test", "/"));
+  EXPECT_TRUE(NavigateIframeToURL(web_contents(), "test", iframe_url));
+
+  auto* iframe_rfh = ChildFrameAt(web_contents()->GetMainFrame(), 0);
+
+  for (size_t i = 0; i < sizeof(cases) / sizeof(cases[0]); ++i) {
+    // Will return empty responses without prompt.
+    BraveWalletPermissionContext::RequestPermissions(
+        cases[i].type, iframe_rfh, cases[i].addresses,
+        base::BindOnce([](const std::vector<ContentSetting>& responses) {
+          EXPECT_FALSE(responses.empty());
+        }));
+
+    content::RunAllTasksUntilIdle();
+    EXPECT_FALSE(IsPendingGroupedRequestsEmpty(cases[i].type)) << "case: " << i;
```

Comment:
can also write it like this, seems okay to distinguish cases by the wallet address:
```
for (const auto& case : cases) {
  SCOPED_TRACE(testing::PrintToString(case.addresses));
  ...
```

Reply:
I didn't know about that, thanks! 
Fixed in 9d0ff1f736f79bae8aed8eaf90607b24469e8d6b